### DISCUSSION TITLE
Responses from createHandler in express are not returned

### DISCUSSION THREAD

**Discussion by mjmahone** [graphql/graphql-http#75] [Q&A] [answered]
I tried to implement the code sample for `express` in order to migrate away from `express-graphql`. The `express-graphql` version works and returns a response, but the `http-graphql` version does not respond.

Code:
```
import * as express from 'express';
import { graphqlHTTP } from 'express-graphql';
import { GraphQLObjectType, GraphQLSchema, GraphQLString } from 'graphql';
import { createHandler } from 'graphql-http/lib/use/express';

const schema = new GraphQLSchema({
  query: new GraphQLObjectType({
    name: "Query", 
    fields: {hello: {type: GraphQLString, resolve: () => "hello"}}
  })
});
const app = express();

// express-graphql implementation
app.use('/graphql', graphqlHTTP({ schema: schema }));

// graphql-http express implementation
app.all('/graphql2', createHandler({ schema }));
app.listen(4000);
```

The I use curl:
```
curl http://localhost:4000/graphql --data-urlencode "query=query Q { hello }"
{"data":{"hello":"hello"}}

curl http://localhost:4000/graphql2 --data-urlencode "query=query Q { hello }"
```

**Expected Behaviour**
A response of `{"data":{"hello":"hello"}}` for both `/graphql` and `/graphql2`.

**Actual Behaviour**
Only `/graphql` (the express-graphql implementation) returns *any* response.

**Version: 1.17.1**
**Node version: 16.14.0**

**Comment by mjmahone** [comment]
Looks like the issue is not express specific: I have the same problem for `import { createHandler } from 'graphql-http/lib/use/http';`

**Comment by mjmahone** [comment]
Weirdly, things work with `ts-node`, but when I `npm run tsc` and then `node dist/main.js` it does not.

**Comment by enisdenjo** [comment, marked as answer]
If you append the flag `-i` to the curl request to check the headers you'll get:
```sh
$ curl http://localhost:4000/graphql2 --data-urlencode "query=query Q { hello }" -i

HTTP/1.1 415 Unsupported Media Type
X-Powered-By: Express
Date: Mon, 10 Apr 2023 07:49:22 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```

This is because POST with `application/x-www-form-urlencoded` body is not supported. [Only with GET.](https://graphql.github.io/graphql-over-http/draft/#sec-GET)

To have curl encode the data to the URL, you have to add the `-G` flag.

```sh
$ curl http://localhost:4000/graphql2 --data-urlencode "query=query Q { hello }" -G

{"data":{"hello":"hello"}}
```
